packages/flame/lib/src/components/text_component.dart

```diff
@@ -39,13 +39,13 @@ class TextComponent<T extends TextRenderer> extends PositionComponent {
 
   void _updateBox() {
     final size = textRenderer.measureText(_text);
-    size.setValues(size.x, size.y);
+    this.size.setValues(size.x, size.y);
   }
 
   @mustCallSuper
   @override
   void render(Canvas canvas) {
     super.render(canvas);
-    _textRenderer.render(canvas, text, Vector2.zero(), anchor: anchor);
+    _textRenderer.render(canvas, text, Vector2.zero());
```

Comment:
prepareCanvas is responsible for handling anchors

Reply:
@spydon recently change that line with the exactly opposite change that you did, his change fixed a bug

Reply:
you mean adding the anchor? I think he introduced a bug. `prepareCanvas` is responsible for doing the anchoring. this is now working. he probably was confused because the size was zero due to another unrelated bug.

Reply:
Yeah, the anchor, I am pretty sure it fixed something, at least some user confirmed that adding anchor here fixed it, this is the PR https://github.com/flame-engine/flame/pull/817

But either way I am just raising awareness to this, I don't know much about the issue, better see what @spydon thinks

Reply:
yeah the issue was that the size was zero, that was the wrong fix, it now causes the anchor to be double applied. I didn't noticed it back then, but that is totally incorrect.

Reply:
Ahhhhh that makes sense! Again, damn shadowing!!!!!!

Reply:
True, my bad!

Reply:
np, I totally missed it too, it seemed to make sense at first glance until you analyze that it gets prepareCanvas from the parent class.